crates/storage/src/lazy/lazy_imap.rs

```diff
@@ -323,7 +323,7 @@ where
     /// a `*mut Entry<T>` pointer that allows for exclusive access. This is safe
     /// within internal use only and should never be given outside the lazy entity
     /// for public `&self` methods.
-    unsafe fn lazily_load(&self, index: Index) -> NonNull<StorageEntry<V>> {
+    pub(crate) unsafe fn lazily_load(&self, index: Index) -> NonNull<StorageEntry<V>> {
```

Comment:
So based off the doc comments it seems like we shouldn't be exposing this. You'll probably need to move the `ContiguousStorage` implementation to this module as a workaround